reflection_refraction.py

```diff
@@ -0,0 +1,136 @@
+from distutils.util import strtobool
+import math
+from matplotlib import pyplot as plt
+
+distance = 5
+
+# Introduction
+def intro():
+    print("\nReflection and refraction of light Calculator")
+    print("By Martin Lambov")
+    print("GitHub: https://github.com/5neepy")
+
+# Check if the equetion is posible
+def check_if_posible(alfa, n1, n2):
+    if alfa >= 90 or alfa <= 0:
+        print("\nAlfa can't be equal or more than 90 and can't be equal or less than 0.\n")
+        exit()
+    if n1 == n2:
+        print("\nIf n_1 = n_2 we don't have Reflection/Refraction of light.\n")
+        exit()
+
+def get_vars():
+    alfa = float(input("\nEnter the degrees of alfa: "))
+    check_if_posible(alfa, 0, 1)
+
+    print("\nEnter the indexes of refraction of usual objects: \n air = 1 \n water = 1.33 \n glass = 1.5")
+
+    n1 = float(input("\nn1: "))
+    n2 = float(input("n2: "))
+    check_if_posible(alfa, n1, n2)
+
+    return alfa, n1, n2
+
+def calculate_refl_angl(alfa, n1, n2):
+    alfa_prim = alfa
+    
+    # Check if we have total internal reflection
+    tot_intern_refl = lambda theta, n1, n2: theta >= 45 and n1 > n2
+    
+    if tot_intern_refl(alfa, n1, n2):
+        beta = 90
+    else:
+        beta = math.degrees(math.asin(n1*math.sin(math.radians(alfa))/n2))  # n1*sin(θ1) = n2*sin(θ2)
+    
+    return alfa, alfa_prim, beta
+
+# Defind function Reflection Plot 
+def calc_ref_plot(alfa, beta):
+    # Incident beam = a:
+    alfa_2 = 90+alfa
+    X_1 = (0+distance*math.cos(math.radians(alfa_2)))
+    Y_1 = (0+distance*math.sin(math.radians(alfa_2)))
+
+    # Reflected beam = a_1:
+    X_2 = -X_1
+    Y_2 = Y_1
+
+    # Refracted beam = b:
+    beta_2 = 270+beta
+    X_3 = (0+distance*math.cos(math.radians(beta_2)))
+    Y_3 = (0+distance*math.sin(math.radians(beta_2)))
+
+    p1 = [X_1, Y_1]
+    p2 = [X_2, Y_2]
+    p3 = [X_3, Y_3]
+
+    return p1, p2, p3
+
+def matplotlib_graph(p1, p2, p3, n1, n2):
+    
+    # plt.title("")
+
+    # Cord. Sys.
+    ax = plt.gca()
+    ax.spines['top'].set_color('none')
+    ax.spines['bottom'].set_position('zero')
+    ax.spines['left'].set_position('zero')
+    ax.spines['right'].set_color('none')
+
+    # Set Cord. sys. Range and making 1x=1y
+    ax.set_xlim([-distance, distance])
+    ax.set_ylim([-distance, distance])
+    plt.axis('equal')
+
+    # Hide numbers
+    plt.gca().axes.get_xaxis().set_ticks([])
+    plt.gca().axes.get_yaxis().set_ticks([])
+
+    # TODO: Print angles
+
+    # With Vectors
+    # color codes: #ff0a0a; #ff4141
```

Comment:
Better define constants with the color codes instead of using comments.

Reply:
Ok